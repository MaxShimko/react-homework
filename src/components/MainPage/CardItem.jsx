import React from 'react';

import '../../styles/CardItem.scss';

class CardItem extends React.PureComponent {
    render() {
        return <div className="card-item">
            <img src={this.props.posterUrl} />
            <div className="delimiter-1"></div>
            <div className="description">
                <div className="left">
                    <h2>{this.props.title}</h2>
                    <p>{this.props.ganre}</p>
                </div>
                <div className="right">
                    <div className="year">{this.props.year}</div>
                </div>
            </div>
        </div>;
    }
}

export default CardItem;
