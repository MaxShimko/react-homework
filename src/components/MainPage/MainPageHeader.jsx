import React from 'react';
import { Container, Row, Col } from 'react-bootstrap';

import '../../styles/MainPageHeader.scss';
import Search from "../Search.jsx";

export default function MainPageHeader({onSearch}) {
    return (
        <div className="main-page-header">
            <Container>
                <Row>
                    <span className="logotype">
                        <span className="bold">netflix</span>roulette
                    </span>
                </Row>
                <div className="delimiter-6" />
                <Row>
                    <Col md={{span: 10, offset: 1}}>
                        <h1>Find your movie</h1>
                        <div className="delimiter-2" />
                        <Search onSearch={onSearch.bind(this)} />
                    </Col>
                </Row>
                <div className="delimiter-10" />
            </Container>
        </div>
    );
};
