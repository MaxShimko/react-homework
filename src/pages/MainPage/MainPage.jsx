import React from 'react';
import {Col, Container, Row} from "react-bootstrap";

import './style.scss';
import Header from './components/Header/Header.jsx';
import CardsContainer from '../../components/CardsContainer/CardsContainer.jsx';
import Footer from '../../components/Footer/Footer.jsx';
import Toggler from "../../components/Toggler/Toggler.jsx";

class MainPage extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            filter: {type: 0, string: ''},
            sortType: 0
        };
    }

    onSearch(filter) {
        this.setState({
           filter
        });
    }

    onSortTypeChange(id) {
        this.setState({
            sortType: id
        });
    }

    render() {
        return (
            <div className="main-page">
                <Header onSearch={this.onSearch.bind(this)} />
                <Container>
                    <div className="delimiter-2" />
                    <Row className="justify-content-end">
                        <Col xs="auto">
                            <Toggler title="Sort by" tabs={['Release date', 'Rating']} onChange={this.onSortTypeChange.bind(this)} />
                        </Col>
                    </Row>
                    <div className="delimiter-2" />
                </Container>
                <CardsContainer filter={this.state.filter} sortType={this.state.sortType} />
                <Footer />
            </div>
        );
    }
}

export default MainPage;
