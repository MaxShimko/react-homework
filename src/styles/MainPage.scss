.main-page {
  background-color: #555;
}
