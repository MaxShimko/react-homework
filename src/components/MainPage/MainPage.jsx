import React from 'react';

import '../../styles/MainPage.scss';
import MainPageHeader from './MainPageHeader.jsx';
import CardsContainer from '../CardsContainer.jsx';
import Footer from '../Footer.jsx';
import {Col, Container, Row} from "react-bootstrap";
import Toggler from "../Toggler.jsx";

class MainPage extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            filter: {type: 0, string: ''},
            sortType: 0
        };
    }

    onSearch(filter) {
        this.setState({
           filter
        });
    }

    onSortTypeChange(id) {
        this.setState({
            sortType: id
        });
    }

    render() {
        return (
            <div className="main-page">
                <MainPageHeader onSearch={this.onSearch.bind(this)} />
                <Container>
                    <div className="delimiter-2" />
                    <Row className="justify-content-end">
                        <Col xs="auto">
                            <Toggler title="Sort by" tabs={['Release date', 'Rating']} onChange={this.onSortTypeChange.bind(this)} />
                        </Col>
                    </Row>
                    <div className="delimiter-2" />
                </Container>
                <CardsContainer filter={this.state.filter} sortType={this.state.sortType} />
                <Footer />
            </div>
        );
    }
}

export default MainPage;
