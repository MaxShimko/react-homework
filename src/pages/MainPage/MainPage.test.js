import React from 'react';
import {shallow} from 'enzyme';
import MainPage from './MainPage';
import {describe, expect, it} from '@jest/globals';
import toJson from 'enzyme-to-json';

describe('<MainPage />', () => {
    it('renders without crashing', () => {
        const component = shallow(<MainPage/>);
        expect(toJson(component)).toMatchSnapshot();
    });
});
