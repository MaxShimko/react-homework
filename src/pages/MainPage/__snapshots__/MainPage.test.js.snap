// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainPage /> renders without crashing 1`] = `
<div
  className="main-page"
>
  <Header
    onSearch={[Function]}
  />
  <Container
    fluid={false}
  >
    <div
      className="delimiter-2"
    />
    <Row
      className="justify-content-end"
      noGutters={false}
    >
      <Col
        xs="auto"
      >
        <Toggler
          onChange={[Function]}
          tabs={
            Array [
              "Release date",
              "Rating",
            ]
          }
          title="Sort by"
        />
      </Col>
    </Row>
    <div
      className="delimiter-2"
    />
  </Container>
  <CardsContainer
    filter={
      Object {
        "string": "",
        "type": 0,
      }
    }
    sortType={0}
  />
  <Footer />
</div>
`;
